tools/checkstyle/checkstyle.bzl

```diff
@@ -0,0 +1,191 @@
+      "commons_collections_commons_collections": "3.2.2",
+      "com_google_guava_guava23": "23.0",
+      "org_slf4j_slf4j_api": "1.7.7",
+      "org_slf4j_slf4j_jcl": "1.7.7",
+    }
+):
+  if not "antlr_antlr" in omit:
+    native.maven_jar(
+        name = "antlr_antlr",
+        artifact = "antlr:antlr:" + versions["antlr_antlr"],
+    )
+  if not "org_antlr_antlr4_runtime" in omit:
+    native.maven_jar(
+        name = "org_antlr_antlr4_runtime",
+        artifact = "org.antlr:antlr4-runtime:" + versions["org_antlr_antlr4_runtime"],
+    )
+  if not "com_puppycrawl_tools_checkstyle" in omit:
+    native.maven_jar(
+        name = "com_puppycrawl_tools_checkstyle",
+        artifact = "com.puppycrawl.tools:checkstyle:" + versions["com_puppycrawl_tools_checkstyle"],
+    )
+  if not "commons_beanutils_commons_beanutils" in omit:
+    native.maven_jar(
+        name = "commons_beanutils_commons_beanutils",
+        artifact = "commons-beanutils:commons-beanutils:" + versions["commons_beanutils_commons_beanutils"],
+    )
+  if not "info_picocli_picocli" in omit:
+    native.maven_jar(
+        name = "info_picocli_picocli",
+        artifact = "info.picocli:picocli:" + versions["info_picocli_picocli"],
+    )
+  if not "commons_collections_commons_collections" in omit:
+    native.maven_jar(
+        name = "commons_collections_commons_collections",
+        artifact = "commons-collections:commons-collections:" + versions["commons_collections_commons_collections"],
+    )
+  if not "com_google_guava_guava23" in omit:
+    native.maven_jar(
+        name = "com_google_guava_guava23",
+        artifact = "com.google.guava:guava:" + versions["com_google_guava_guava23"],
+    )
+  if not "org_slf4j_slf4j_api" in omit:
+    native.maven_jar(
+        name = "org_slf4j_slf4j_api",
+        artifact = "org.slf4j:slf4j-api:" + versions["org_slf4j_slf4j_api"],
+    )
+  if not "org_slf4j_slf4j_jcl" in omit:
+    native.maven_jar(
+        name = "org_slf4j_slf4j_jcl",
+        artifact = "org.slf4j:jcl-over-slf4j:" + versions["org_slf4j_slf4j_jcl"],
+    )
+
+
+JavaSourceFiles = provider(
+    fields = {
+        'files' : 'java source files'
+    }
+)
+
+
+def collect_sources_impl(target, ctx):
+    files = []
+    if hasattr(ctx.rule.attr, 'srcs'):
+        for src in ctx.rule.attr.srcs:
+            for f in src.files:
+                if f.extension == 'java':
+                    files.append(f)
+    return [JavaSourceFiles(files = files)]
+
+
+collect_sources = aspect(
+    implementation = collect_sources_impl,
+)
+
+
+def _checkstyle_test_impl(ctx):
+    if "{}-checkstyle".format(ctx.attr.target.label.name) != ctx.attr.name:
+        fail("target should follow `{java_library target name}-checkstyle` pattern")
+    properties = ctx.file.properties
+    suppressions = ctx.file.suppressions
+    opts = ctx.attr.opts
+    sopts = ctx.attr.string_opts
+
+    classpath = ":".join([file.path for file in ctx.files._classpath])
+
+    args = ""
```

Comment:
Please thoroughly document this method, as it's _not_ trivial.

Reply:
@oferb should be fixed in https://github.com/google/startup-os/pull/486/commits/70e8e9de1d93cdafea77bd4488c6318c5c648a36
if it's not, please pinpoint what should be described more thoroughly 🙂